WalletWasabi/Wallets/WalletDirectories.cs

```diff
@@ -68,19 +68,54 @@ public IEnumerable<FileInfo> EnumerateWalletFiles(bool includeBackupDir = false)
 			return result.OrderByDescending(t => t.LastAccessTimeUtc);
 		}
 
-		public string GetNextWalletName(string prefix = "Wallet")
+		public string GetNextWalletName(string prefix = "Random Wallet")
 		{
-			int i = 0;
+			int i = 1;
+			var walletNames = EnumerateWalletFiles(true).Select(x => Path.GetFileNameWithoutExtension(x.Name));
 			while (true)
 			{
-				var walletName = $"{prefix}{i}";
-				if (!File.Exists(GetWalletFilePaths(walletName).walletFilePath))
+				var walletName = i == 1 ? prefix : $"{prefix} {i}";
+
+				if (!walletNames.Contains(walletName))
 				{
 					return walletName;
 				}
 
 				i++;
 			}
 		}
+
+		public string GetNextHardwareWalletName(HardwareWalletModels model)
+		{
+			string prefix;
+			if (model == HardwareWalletModels.Unknown)
+			{
+				prefix = "Hardware Wallet";
+			}
+			else if (model == HardwareWalletModels.BitBox02_BTCOnly
+				|| model == HardwareWalletModels.BitBox02_Multi
+				|| model == HardwareWalletModels.DigitalBitBox_01)
+			{
+				prefix = "BitBox";
+			}
+			else if (model == HardwareWalletModels.DigitalBitBox_01_Simulator)
+			{
+				prefix = "BitBox Simulator";
+			}
+			else if (model == HardwareWalletModels.Trezor_1)
+			{
+				prefix = "Trezor One";
+			}
+			else if (model == HardwareWalletModels.Trezor_1_Simulator)
+			{
+				prefix = "Trezor One Simulator";
+			}
+			else
+			{
+				prefix = GetNextWalletName(model.ToString().Replace('_', ' '));
+			}
+
+			return GetNextWalletName(prefix);
+		}
```

Comment:
What about `Description("Friendly name")` tag?

For example:
![image](https://user-images.githubusercontent.com/16364053/93335883-59aacc00-f827-11ea-9098-d43004d4f776.png)

And later:
![image](https://user-images.githubusercontent.com/16364053/93336036-83fc8980-f827-11ea-9c57-602de681d6a3.png)

Reply:
@soosr Sounds good to me. Can you take this PR over from me?

Reply:
yepp